### PR TITLE
Implement sync+freeze embed/LM-head tying for early training

### PR DESCRIPTION
**Pull request by ChinmayK0607**
- Initialize wte from lm_head weights
- During tied phase: zero wte grads so only lm_head updates
- After optimizer step: sync wte weights from lm_head each step
- Add split point (split-embed-frac) to stop tying after ~67% of training

val loss: 3.382950
time: 14.60m

wandb log: https://wandb.ai/ChinmayK0604/nanochat/runs/utoeyg21/logs

tested with seed variance. 

Test methodology: 

do multiple seeded runs(atleast 3) and compare val loss, mean, spread, difference etc. 
if difference is consistent and does not exceed the baseline val loss, confirm as a valid improvement. 

attaching image of two runs that i did just to do a smoke test before 3 run benchmark. 
image

happy to help!

### CODE REVIEW

**Comment by akshayvegesna** [comment]
Yeah, seems best to hold off on this one
